src/types/list.md

```diff
@@ -1 +1,66 @@
 # List
+
+Lists are the SML type of ordered collections of objects. Notably, you can create lists of any kind of object, so `int list`, `string list`, and `bool list` are all valid types. `list` on its own is a type constructor (meaning that it makes new types out of old types), so it is not a valid type by itself, however. Lists are not, however, the same as arrays - they do not have constant-time access to any given index of the list. Instead, you only have access to the elements located at the very beginning of the list, the _head_. Additionally, lists are more _restricted_ than data structures in some other languages - a given list has a fixed type for its elements. All of the elements in a list must be of the same type. For the purposes of this document, we will discuss only int lists.
+
+## Values
+
+We write lists as a sequence of integers, separated by commas, all enclosed with two square brackets. So then we have valid int lists as `[1, 2, 3]`, `[1, 5, 1, 5, 0]`, and `[]`, with the latter representing the empty list. We also refer to the empty list as _nil_ (in addition, you can type `nil` instead of `[]` in code). 
+
+The other essential component to lists is what is known as the `::` operator, referred to as "cons".
```

Comment:
Related to the above, I think it's a bit misleading to say that the type of cons is specialized to ints. It might be a better idea to describe its type as `t * t list -> t list` or something (for any type `t`).